lib/bali/concerns/date_range_attribute.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+# Allows to define date range attributes.
+# It will convert '2023-01-01 to 2023-01-10' to a Range
+#
+# Example:
+#
+#    class Model
+#      include ActiveModel::Attributes
+#      include Bali::Concerns::DateRangeAttribute
+#
+#      date_range_attribute :date_range, default: Time.zone.now.all_day
+#
+#    end
+#
+
+module Bali
+  module Concerns
+    module DateRangeAttribute
+      extend ActiveSupport::Concern
+
+      class_methods do
+        # @param name [Date|String] The name of the attribute
+        # @param default [Range|String] The default value for the attribute
+        # @param start_attribute [String|Symbol] Attribute to store the start of the range
+        # @param end_attribute [String|Symbol] Attribute to store the end of the range
+
+        def date_range_attribute(name, default: nil, start_attribute: nil, end_attribute: nil)
+          attribute(name, default: default)
+
+          [start_attribute, end_attribute].compact.each do |attribute|
+            attribute(attribute) unless respond_to?(attribute)
+          end
```

Comment:
No esta mas sencillo asi?

```suggestion
          attribute :start_attribute if start_attribute.present? && !respond_to? :start_attribute
          attribute :end_attribute if end_attribute.present? && !respond_to? :end_attribute
```